lib/carrierwave/locale/en.yml

```
@@ -9,5 +9,6 @@ en:
       content_type_allowlist_error: "You are not allowed to upload %{content_type} files, allowed types: %{allowed_types}"
       content_type_denylist_error: "You are not allowed to upload %{content_type} files"
       processing_error: "Failed to manipulate, maybe it is not an image?"
+      rmagick_error_security_policy: "Failed to manipulate due to imagemagick's security policy for this filetype"
```

How about an error message like this?

```suggestion
      security_policy_error: "Uploading this format of files is not allowed due to security reasons. Try different one"
```

This will be given to end-users as a validation message, so a word like `imagemagick` is useless to them.

---

I'd prefer it if that error message would help the user to find the cause of it as fast as possible. "Imagemagick" and "security policy" are must-have keywords for this.

---

And "uploading this format of files is no allowed" is misleading because such cases should be covered by the code at an earlier point.

---

> such cases should be covered by the code at an earlier point

I don't think so. Having a specific file extension or content type and actually being in some file format is different things.

Then how about re-raising the original exception? In that way you can know something happened with ImageMagick. I can't merge this PR as it is, since it goes against the intention of #2499.

---

I'm not interested in your _wasted_ hours. I'm maintaining this library for free, can I call this conversation waste of time then?
Let's focus on how to make this library better, please.

My point is that showing the word `imagemagick` to end-users as the validation message is useless and confusing, as stated in #2499. What's useful is to let end-users know what to do next. What do you want them to do? Try another file, or consult with the administrator?